byzcoin/service.go

```diff
@@ -1012,6 +1037,55 @@ func (s *Service) createNewBlock(scID skipchain.SkipBlockID, r *onet.Roster, tx
 	return ssbReply.Latest, nil
 }
 
+// createUpgradeVersionBlock has the sole purpose of proposing an empty block with the
+// version field of the DataHeader updated so that new blocks will use the new version,
+func (s *Service) createUpgradeVersionBlock(scID skipchain.SkipBlockID, version Version) (*skipchain.SkipBlock, error) {
```

Comment:
when is this function called in practice? i only see it in the test.

in a later PR are you planning to add perhaps a bcadmin feature that will allow a client to ask the leader to send a version upgrade transaction? is this the version upgrade mechanism?

Reply:
Yes for the moment only in the test, that was for demonstration. A PR is on the way to let the leader detect when other conodes are up-to-date by using the collectTx protocol.